CapstoneProject/Assets/Scripts/Core/Effect/CharacterEffect.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = System.Random;
+using Cinemachine;
+using Core.Components;
+using TMPro;
+
+[RequireComponent(typeof(RigidbodyComponent))]
+public class CharacterEffect : MonoBehaviour
+{
+    [SerializeField]
+    private SpriteRenderer _spriteRenderer;
+    public float flashDuration = 0.2f;
+    [SerializeField]
+    private Material _hitMat;
+    
+    private Material _originalMat;
+
+    [Header("Screen shake FX")]
+    [SerializeField]
+    private CinemachineImpulseSource _screenShake;
+    [SerializeField]
+    private float _shakeMultiplier;
+    [SerializeField]
+    private Vector3 _shakePower;
+    
+    [Header("Hit FX")]
+    [SerializeField] private GameObject _hitFX01Prefab;
+    [SerializeField] private GameObject _hitFX02Prefab;
+
+    /*[Header("Slash FX")]
+    [SerializeField] private GameObject _slashFX01Prefab;
+    [SerializeField] private GameObject _slashFX02Prefab;*/
+    
+
+    [SerializeField]
+    private RigidbodyComponent _rb;
+    private void Start()
+    {
+        _originalMat = _spriteRenderer.material;
+    }
+
+    public IEnumerator FlashFX()
+    {
+        _spriteRenderer.material = _hitMat;
+        yield return new WaitForSeconds(flashDuration);
+        _spriteRenderer.material = _originalMat;
+    }
+
+    public void BlinkRedColor()
+    {
+        if (_spriteRenderer.color != Color.white)
+            _spriteRenderer.color = Color.white;
+        else
+            _spriteRenderer.color = Color.red;
+    }
+
+    public void CancelRedBlink()
+    {
+        CancelInvoke();
+        _spriteRenderer.color = Color.white;
+    }
+
+    public void ShakeScreen()
+    {
+        int shakingDirection = _rb.CurrentFacingDirection == RigidbodyComponent.FacingDirections.RIGHT ? 1 : -1;
+        _screenShake.m_DefaultVelocity =
+            new Vector3(_shakePower.x * shakingDirection, _shakePower.y) * _shakeMultiplier;
+        _screenShake.GenerateImpulse();
+    }
+
+    public void GenerateHitFX(Transform targetTransform)
+    {
+        float xPosition = UnityEngine.Random.Range(-0.5f, 0.5f);
+        float yPosition = UnityEngine.Random.Range(-0.5f, 0.5f);
+        float zRotation = UnityEngine.Random.Range(-90, 90);
+        int whichHitFX = UnityEngine.Random.Range(0, 10);
```

Comment:
10 here maybe could be a const for number of effects? is that was 10 represents here?